sparkly/catalog.py

```diff
@@ -26,6 +27,62 @@ def __init__(self, spark):
         """
         self._spark = spark
 
+    def create_table(self, table_name, path=None, source=None, schema=None, **options):
+        """Create table in the metastore.
+
+        Extend SparkSession.Catalog.createExternalTable by accepting
+        a mode='overwrite' option which creates the table even if a
+        table with the same name already exists. All other args are
+        exactly the same.
+
+        Notes:
+            If the table exists, create two unique names, one for the
+            new and one for the old instance, then try to swap names
+            and drop the "old" instance. If any step fails, the metastore
+            might be currently left at a broken state.
+
+        Args:
+            mode (str): if set to 'overwrite', drop any table of the same
+                name from the metastore. Given as a kwarg. Default is
+                error out if table already exists.
+
+        Returns:
+            pyspark.sql.DataFrame, representing the created table.
+        """
+        overwrite_existing_table = (
+            options.pop('mode', '').lower() == 'overwrite' and
+            self.has_table(table_name)
+        )
+
+        def _append_unique_suffix(*args):
+            return '__'.join(args + (uuid.uuid4().hex, ))
+
+        if overwrite_existing_table:
+            new_table_name = _append_unique_suffix(table_name, 'new')
+        else:
+            new_table_name = table_name
+
+        try:
+            createTable = self._spark.catalog.createTable
+        except AttributeError:  # before Spark 2.2
```

Comment:
- it isn't safe, `AttributeError` could be raised at `self._spark` or `self._spark.catalog`
- pythonic way will be to check for `hasattr(self._spark.catalog, 'createTable')`